tools/build/sprite/sprite_shading_profiles.py

```diff
@@ -0,0 +1,167 @@
+            SpriteShadingGroup(
+                area=group["area"],
+                profiles=profiles,
+            )
+        )
+    return groups
+
+def build(input: Path, bin_out: Path, header_out: Path, endian: Literal["big", "little"]="big",
+          matching: bool = True):
+
+    END = ">" if endian == "big" else "<"
+
+    with open(input, "r") as f:
+        json_data = json.load(f)
+
+    groups = groups_from_json(json_data)
+
+    # Header creation
+    with open(header_out, "w") as f:
+        f.write("#ifndef SHADING_PROFILES_H\n")
+        f.write("#define SHADING_PROFILES_H\n")
+        f.write("/* This file is auto-generated from sprite_shading_profiles.json. Do not edit. */\n\n")
+        f.write("enum ShadingProfile {\n")
+        f.write("    SHADING_NONE = 0xFFFFFFFF,\n")
+
+        for g, group in enumerate(groups):
+            for i, profile in enumerate(group.profiles):
+                f.write(f"    SHADING_{profile.name.upper()} = 0x{(g << 16) | i:08X},\n")
+        f.write("};\n")
+        f.write("#endif // SHADING_PROFILES_H")
+
+    data_offset = 0
+    profile_list_offset = len(groups) * 8
+    offsets_table = bytearray()
+    profile_lists = bytearray()
+    data_table = bytearray()
+    for g, group in enumerate(groups):
+        offsets_table.extend(struct.pack(END + "i", data_offset))
+        offsets_table.extend(struct.pack(END + "i", profile_list_offset))
+
+        group_data_offset = 0
+
+        for profile in group.profiles:
+            profile_lists.extend(struct.pack(END + "i", group_data_offset))
+            data_table.extend(struct.pack(END + "B", len(profile.lights)))
+            data_table.append(0)
+            data_table.extend(struct.pack(END + "B", profile.ambient[0]))
+            data_table.extend(struct.pack(END + "B", profile.ambient[1]))
+            data_table.extend(struct.pack(END + "B", profile.ambient[2]))
+            data_table.extend(struct.pack(END + "B", profile.power))
+            group_data_offset += 6
+            profile_list_offset += 4
+
+            for light in profile.lights:
+                data_table.extend(struct.pack(END + "B", light.flags))
+                data_table.extend(struct.pack(END + "B", light.rgb[0]))
+                data_table.extend(struct.pack(END + "B", light.rgb[1]))
+                data_table.extend(struct.pack(END + "B", light.rgb[2]))
+                data_table.extend(struct.pack(END + "h", light.pos[0]))
+                data_table.extend(struct.pack(END + "h", light.pos[1]))
+                data_table.extend(struct.pack(END + "h", light.pos[2]))
+                data_table.extend(struct.pack(END + "f", light.falloff))
+                data_table.append(0)
+                data_table.append(0)
+                group_data_offset += 0x10
+        data_offset += group_data_offset
+
+    offsets_table.extend(profile_lists)
+    if matching:
+        offsets_table += b'\0' * (0x1D0 - len(offsets_table)) # Pad to 0x1D0
```

Comment:
Interesting that this is a thing we have to do, does this imply that we should have a bunch of zero groups in the input config?

Reply:
That would cause more lighting data generation as well. I really think this is just padding until the actual lighting data